src/infrastructure/healthchecks/HealthcheckRefreshJob.ts

```diff
@@ -0,0 +1,33 @@
+import type { CommonDependencies } from '../commonDiConfig.js'
+import { AbstractPeriodicJob } from '../jobs/AbstractPeriodicJob.js'
+import type { HealthcheckClass, SupportedHealthchecks } from './healthchecks.js'
+
+export class HealthcheckRefreshJob extends AbstractPeriodicJob {
+  public static JOB_NAME = 'HealthcheckRefreshJob'
+  private readonly healthCheckers: Record<SupportedHealthchecks, HealthcheckClass>
+
+  constructor(dependencies: CommonDependencies) {
+    super(
+      {
+        jobId: HealthcheckRefreshJob.JOB_NAME,
+        intervalInMs: 10000,
+        singleConsumerMode: {
+          enabled: true,
+        },
+      },
+      dependencies,
+    )
+
+    this.healthCheckers = {
```

Comment:
🟠 @kibertoad should we make this a parameter and set the jobs in di config instead? I don't think the refresh job needs to know which ones it is refreshing, does it?

Reply:
good point, will adjust

Reply:
improved